xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm

```diff
@@ -129,7 +129,42 @@
 #if($xcontext.action == "admin" || $xcontext.action == "import" || $xcontext.action == "export")
   #set($bodyAction = "view")
 #end
+## Query user preferences for UI customization
+## Get the legacy preference for backward compatibility
+#set ($a11y = "$!{xwiki.getUserPreference('accessibility')}")
```

Comment:
Is there any code to clear that legacy preference? I wonder if the code shouldn't consider this setting only if the user preferences have never been saved with the new settings and make sure that the user profile takes the old value into account for initializing the preference.

Reply:
> Is there any code to clear that legacy preference?

Not as of now. I decided to leave it in so that nothing changes for users if they never update their accessibility preferences.

> I wonder if the code shouldn't consider this setting only if the user preferences have never been saved with the new settings and make sure that the user profile takes the old value into account for initializing the preference.

When the user updates preferences, we can't be sure that the accessibility ones are the ones updated. I don't think we want to fill in the accessibility fields with values different from default when the user edits its default editor ... IMO this would be a pretty deceptive pattern and leaving the old value in makes it easy both user experience and for development. We just need to set a note in documentation about this difference in behaviour in between seemingly and things should be okay.